### PR TITLE
addition and updates to CI workflows

### PR DESCRIPTION
**Pull request by smallTrogdor**
#### TL;DR
* add release workflow
* update test workflow

####
With this PR, there are two updates being introduced to our workflow.

First, a new Design System Flutter Release workflow is introduced: The workflow is manually triggered with _patch_, _minor_ or _major_ argument and turns the [Unreleased] section of the CHANGELOG.md into a tag, an automatic commit and a consequent github release fully automatic.

Second, the test workflow is triggered **on every PR to main** (before on every push to any branch) and includes a validation of the CHANGELOG.md schema defined in [Keep A Changelog](https://keepachangelog.com/en/1.1.0/). Please note that this includes the allowed sub sections in the version releases (e.g. a `Removed` section **cannot** be included in a _patch_ or _minor_ version release).

We aim to further go into the **GitHub Flow** direction - i.e. having short lived feature branches with the addition of fully controlled releases (slower).

### CODE REVIEW

**Comment by smallTrogdor** [comment]
I also updated the CHANGELOG.md file to have all the GitHub releases from previous releases.

**Comment by smallTrogdor** [comment]
superseded by #118
